app/controllers/users/two_factor_verification_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Users
+  class TwoFactorVerificationController < ApplicationController
+    before_action :check_direct_otp_expiry, only: [:show, :create]
+
+
+    def show
+      unless current_user.totp_enabled?
+        current_user.create_direct_otp
+        r = SMSService.new.send_two_factor_confirmation_code!(current_user)
+
+        # TODO: make this async and handle service failure
+        unless r.code == 202
+          raise 'There was an error sending the SMS'
```

Comment:
We probably do - but do we need to handle this in any sophisticated manner? If the SMS API is (intermittently or otherwise) down, do we want a strategy of retrying? If not, we could probably settle for displaying an informative message of `The 2FA service is currently down`, but would need to test how helpful that is.